### ISSUE TITLE
BGC diagnostics are zero in some tiles inconsistently

### ISSUE THREAD

**Issue by ubbu36** [CESR-lab/ucla-roms#12]
I'm adding this as an issue for visibility.

I'm having problems where the bgc_avg_dia output is zero in some tiles. This is in turn affecting the Alk and DIC fields.

This seems to be happening in different tiles during different runs, even with the same restart file and same executable. I've attached a few movies here that illustrate the issue in two different runs.

https://github.com/CESR-lab/ucla-roms/assets/69470952/3d693812-963f-40c4-9a9b-0c14642b9379

https://github.com/CESR-lab/ucla-roms/assets/69470952/2918030f-b280-4a22-8d94-3140f291d0cd

**Comment by matt-long** [comment]
@ubbu36, I thought you were also seeing the block structure in differences between tracers in different runs. Is that the case?

**Comment by matt-long** [comment]
All the BGC diagnostics you point to are contingent on the pH solver converging. Is it possible that that the pH solve is failing silently? Why would infect the whole block?

**Comment by ubbu36** [comment]
@matt-long yes, I'm seeing the differences in tracers between runs, but they tracers are not 0, they are just different because, presumably, the BGC is failing. You can only see the difference in tracers if two runs are subtracted like in the video attached here:

https://github.com/CESR-lab/ucla-roms/assets/69470952/5ecc69e6-eb91-44f6-bf6d-1578d596b069

**Comment by ubbu36** [comment]
So far, in the updated version running on Perlmutter, I haven't experienced this issue
